src/cq.rs

```diff
@@ -132,15 +134,45 @@ impl<'a> Drop for CompletionQueueRef<'a> {
     }
 }
 
+pub struct WorkerInfo {
+    id: ThreadId,
+    pending_work: UnsafeCell<VecDeque<UnfinishedWork>>,
+}
+
+unsafe impl Sync for WorkerInfo {}
```

Comment:
`push_work` is safe to be called across threads, only the thread that creates the completion queue can push successfully, otherwise nothing will happen. `pop_work` is indeed unsafe, I will change it to `&mut self` instead.